content/guides/kubernetes/k8s-secrets-sa-what-is.md

```diff
@@ -0,0 +1,257 @@
+
+All examples used in this guide can be [found on GitHub](https://github.com/BrianMMcClain/k8s-secrets-and-sa).
+
+Before you can get started using secrets, you first need to create a secret. As you may expect, this can be done by defining an object of kind `Secret`:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+data:
+  username: bXl1c2VybmFtZQo=
+  password: bXlwYXNzd29yZAo=
+```
+
+Secrets in Kubernetes are, at their most basic form, a collection of keys and values. The above example creates a secret named `mysecret` with two keys: `username` and `password`. There’s one very important thing to note though, which is that the values of these key/value pairs are encoded as base64. Remember that base64 is an encoding algorithm, not an encryption algorithm. This is done to help facilitate data that may not be entirely alpha-numeric, and instead could include binary data, non-ASCII data, etc. You apply can this YAML as you would if you were creating any other Kubernetes object:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/BrianMMcClain/k8s-secrets-and-sa/main/secret-base64.yaml
+```
+
+Once applied, you can see that while you can get the secret with `kubectl`, it avoids printing the values of each key by default:
+
+```bash
+$ kubectl describe secret mysecret
+
+Name:         mysecret
+Namespace:    default
+Labels:       <none>
+Annotations:  
+Type:         Opaque
+
+Data
+====
+password:  11 bytes
+username:  11 bytes
+```
+
+Of course, if you want to see the base64-encoded contents of the secret, you can still fetch them with a slightly different command:
+
+```bash
+$ kubectl get secret mysecret -oyaml
+
+apiVersion: v1
+data:
+  password: bXlwYXNzd29yZAo=
+  username: bXl1c2VybmFtZQo=
+kind: Secret
+...
+```
+Great! With your secret created, it’s time to start creating pods to use it! You’re faced with another decision, however, since Kubernetes provides a couple of methods for presenting secrets to a pod. The first example you’ll look at is mounting them as files in a volume:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: secret-as-file
+spec:
+  containers:
+  - name: secret-as-file
+    image: nginx
+    volumeMounts:
+    - name: mysecretvol
+      mountPath: "/etc/mysecret"
+      readOnly: true
+  volumes:
+  - name: mysecretvol
+    secret:
+      secretName: mysecret
+```
+
+Here, a new pod named `secret-as-file` is created from the [nginx Docker image](https://hub.docker.com/_/nginx). 
+
+> NOTE: The nginx container image is used here simply because it's an easily accesable long-running process, this would look the same for your own container image. 
```

Comment:
```suggestion
> NOTE: The nginx container image is used here simply because it's an easily accessible long-running process, this would look the same for your own container image. 
```